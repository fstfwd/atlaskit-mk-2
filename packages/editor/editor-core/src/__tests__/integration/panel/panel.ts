import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
} from '../_helpers';
import {
  clipboardHelper,
  clipboardInput,
  copyAsHTMLButton,
  copyAsPlaintextButton,
} from '../_helpers';
import { insertMentionUsingClick } from '../message-renderer/_mention-helpers';

BrowserTestCase(
  'Inserts a panel on fullpage',
  { skip: ['edge', 'ie', 'safari'] },
  async client => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);
    await browser.waitForSelector(editable);

    await browser.click(editable);
    await quickInsert(browser, 'Panel');

    // type some text
    await browser.type(editable, 'this text should be in the panel');

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Changes the type of a panel to Error',
  { skip: ['edge', 'ie', 'safari'] },
  async client => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);

    await browser.waitForSelector(fullpage.placeholder);
    await browser.click(fullpage.placeholder);

    await quickInsert(browser, 'Panel');

    // type some text
    await browser.type(editable, 'this text should be in the panel');

    // click on Error label
    const selector = `[aria-label="Error"]`;
    await browser.click(selector);

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Inserts a link into a panel by typing Markdown',
  { skip: ['edge', 'ie', 'safari'] },
  async client => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);

    await browser.waitForSelector(fullpage.placeholder);
    await browser.click(fullpage.placeholder);

    await quickInsert(browser, 'Panel');

    // type some text
    await browser.type(editable, '[Atlassian](https://www.atlassian.com/)');

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

// Cannot paste rich text in IE/Edge
BrowserTestCase(
  'Can paste rich text',
  { skip: ['ie', 'safari', 'edge'] },
  async client => {
    const browser = new Page(client);
    await browser.goto(clipboardHelper);
    await browser.isVisible(clipboardInput);
    await browser.type(
      clipboardInput,
      '<p>this is a link <a href="http://www.google.com">www.google.com</a></p><p>more elements with some <strong>format</strong></p><p>some addition<em> formatting</em></p>',
    );
    await browser.click(copyAsHTMLButton);

    await browser.goto(fullpage.path);
    await browser.waitForSelector(editable);

    await browser.click(editable);
    await quickInsert(browser, 'Panel');

    await browser.paste(editable);

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'can paste plain text',
  { skip: ['ie', 'safari', 'edge'] },
  async client => {
    const browser = new Page(client);
    await browser.goto(clipboardHelper);
    await browser.isVisible(clipboardInput);
    await browser.type(
      clipboardInput,
      'this is a link http://www.google.com more elements with some **format** some addition *formatting*',
    );
    await browser.click(copyAsPlaintextButton);

    await browser.goto(fullpage.path);
    await browser.waitForSelector(editable);

    await browser.click(editable);
    await quickInsert(browser, 'Panel');

    await browser.paste(editable);
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'can insert mention into panel using click',
  { skip: ['ie', 'safari', 'edge'] },
  async client => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);
    await browser.waitFor(editable);
    await browser.click(editable);
    await quickInsert(browser, 'Panel');

    await insertMentionUsingClick(browser, '0');
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
