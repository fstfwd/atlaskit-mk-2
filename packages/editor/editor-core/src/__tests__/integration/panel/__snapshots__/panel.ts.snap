// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can paste rich text 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this is a link ",
              "type": "text",
            },
            Object {
              "marks": Array [
                Object {
                  "attrs": Object {
                    "__confluenceMetadata": null,
                    "href": "http://www.google.com",
                  },
                  "type": "link",
                },
              ],
              "text": "www.google.com",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "content": Array [
            Object {
              "text": "more elements with some ",
              "type": "text",
            },
            Object {
              "marks": Array [
                Object {
                  "type": "strong",
                },
              ],
              "text": "format",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "content": Array [
            Object {
              "text": "some addition",
              "type": "text",
            },
            Object {
              "marks": Array [
                Object {
                  "type": "em",
                },
              ],
              "text": " formatting",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`Changes the type of a panel to Error 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "error",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`Inserts a link into a panel by typing Markdown 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "attrs": Object {
                    "__confluenceMetadata": null,
                    "href": "https://www.atlassian.com/",
                  },
                  "type": "link",
                },
              ],
              "text": "Atlassian",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`Inserts a panel on fullpage 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`can insert mention into panel using click 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "accessLevel": "",
                "id": "0",
                "text": "@Carolyn",
                "userType": null,
              },
              "type": "mention",
            },
            Object {
              "text": " ",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`can paste plain text 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this is a link ",
              "type": "text",
            },
            Object {
              "marks": Array [
                Object {
                  "attrs": Object {
                    "__confluenceMetadata": null,
                    "href": "http://www.google.com",
                  },
                  "type": "link",
                },
              ],
              "text": "http://www.google.com",
              "type": "text",
            },
            Object {
              "text": " more elements with some ",
              "type": "text",
            },
            Object {
              "marks": Array [
                Object {
                  "type": "strong",
                },
              ],
              "text": "format",
              "type": "text",
            },
            Object {
              "text": " some addition ",
              "type": "text",
            },
            Object {
              "marks": Array [
                Object {
                  "type": "em",
                },
              ],
              "text": "formatting",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;
